src/main/java/lessonfour/hometask/exception/UserRegistrationDtoException.java

```diff
@@ -0,0 +1,7 @@
+package lessonfour.hometask.exception;
+
+public class UserRegistrationDtoException extends RuntimeException {
```

Comment:
Not the best exception name. Exception does not relate to the DTO. Exception relates to the business flow. You may have more generic name:
```suggestion
public class UserRegistrationException extends RuntimeException {
```